third_party/terraform/data_sources/data_source_google_compute_image.go

```diff
@@ -121,8 +127,21 @@ func dataSourceGoogleComputeImageRead(d *schema.ResourceData, meta interface{})
 		log.Printf("[DEBUG] Fetching latest non-deprecated image from family %s", v.(string))
 		image, err = config.clientCompute.Images.GetFromFamily(project, v.(string)).Do()
 		log.Printf("[DEBUG] Fetched latest non-deprecated image from family %s", v.(string))
+	} else if v, ok := d.GetOk("filter"); ok {
+		images, err := config.clientCompute.Images.List(project).Filter(v.(string)).Do()
```

Comment:
Hi @upodroid !  Sorry about this, we've been working on some refactoring recently, if you rebase off master you'll see there's a new function in `config.go` called `NewComputeClient`, it will take a string `userAgent` which will already be defined in this function. It creates a new client that replaces the compute client that is currently on the config. Thanks!

```suggestion
		images, err := config.NewComputeClient(userAgent).Images.List(project).Filter(v.(string)).Do()
```

Reply:
Yeah, I saw the module attribution work. 3rd one in 2 two days. I'll get it done

Reply:
fixed